help.py

```diff
@@ -41,3 +42,276 @@ def datetime_to_epoch_millis(date: datetime) -> int:
+
+        if sigma_rules: survey_payload['sigma_rules'] = sigma_rules
+
+        # run search based on IOC file
+        if args.iocfile:
+            with open(args.iocfile) as ioc_file:
+                basename = os.path.basename(args.iocfile)
+                ioc_source = basename
+                ioc_list = [x.strip() for x in ioc_list]
+
+            survey_payload.update({"ioc_source": ioc_source, "ioc_list": ioc_list, "ioc_type": args.ioctype})
+
+        survey_payload['product_args'] = product_arg_builder(args=args)
+
+        survey_payload = {k:v for k,v in survey_payload.items() if v != None}
+        return survey_payload
+    
+def logger(edr:str, logs_dir:str='logs') -> logging.Logger:
+    # instantiate a logger
+    log = logging.getLogger('surveyor')
+    logging.debug(f'Product: {edr}')
+
+    # configure logging
+    root = logging.getLogger()
+    root.setLevel(logging.DEBUG)
+    root.handlers = list()  # remove all default handlers
+    log_format = '[%(asctime)s] [%(levelname)-8s] [%(name)-36s] [%(filename)-20s:%(lineno)-4s] %(message)s'
+
+    # create logging directory if it does not exist
+    os.makedirs(logs_dir, exist_ok=True)
+
+    # create logging file handler
+    log_file_name = datetime.utcnow().strftime('%Y%m%d%H%M%S') + f'.{edr}.log'
+    handler = logging.FileHandler(os.path.join(logs_dir, log_file_name))
+    handler.setLevel(logging.DEBUG)
+    handler.setFormatter(logging.Formatter(log_format))
+    root.addHandler(handler)
+    
+    return log
+
+def check_product_args_structure(edr: str, product_args: dict) -> dict:
+    if product_args == {}:
+        return {"result": True, "edr": edr, "required_fields": "No arguments were supplied"}
+
+    product_check = [k for k, v in product_args.items()]
+    
+    if edr == "cbc":
+        required_fields = ['device_group', 'device_policy']
+    elif edr == "cbr":
+        required_fields = ['sensor_group']
+    elif edr == "s1":
+        required_fields = ['deep_visibility']
+    else:
+        return {"result": False, "edr": edr, "required_fields": required_fields}
+
+    for field in required_fields:
+        if field not in product_check:
+            return {"result": False, "edr": edr, "required_fields": required_fields}
+
+    return {"result": True, "edr": edr, "required_fields": required_fields}
+
+
+def check_credentials_structure(edr, creds: dict) -> dict:
+    # Check credential provided for the bare minimum arguments for execution.
+    # This function checks for the presence of needed arguments; it does not check for valid input.
+    if edr in ["cbc", "cbr"]:
+        return {"result": True, "edr": edr, "required_fields": "Assuming credentials file is stored in required location"}
+    
+    cred_check = [k for k, v in creds.items() if v]
+
+    if 'creds_file' and 'profile' in cred_check:
+        return {"result": True, "edr": edr, "required_fields": ["creds_file", "profile"]}
+
+    if edr == "cbc":
```

Comment:
Thinking this section could also use a global dict at the top of `help.py` to avoid chained if/else logic here